sale_salesperson_filter/views/res_partner_view.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_partner_form" model="ir.ui.view">
+        <field name="name">res.partner.form</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='user_id']" position="replace">
+                <field name="user_id" domain="[('salesperson_select','=', True)]"/>
```

Comment:
Pay attention to this: https://github.com/rfhk/cip-custom/blob/949e7d614b88b0eb8fe222f3f19fc71af55bf385/sale_view_adjust/views/res_partner_views.xml#L4-L13